File: TransCoder_tokenized_test_set_functions/java/FIND_ONE_MULTIPLE_REPEATING_ELEMENTS_READ_ARRAY.java
static int findRepeatingNumber ( int [ ] arr, int n ) {
  int sq = ( int ) Math . sqrt ( n );
  int range = ( n / sq ) + 1;
  int [ ] count = new int [ range ];
  for ( int i = 0;
  i <= n;
  i ++ ) {
    count [ ( arr [ i ] - 1 ) / sq ] ++;
  }
  int selected_block = range - 1;
  for ( int i = 0;
  i < range - 1;
  i ++ ) {
    if ( count [ i ] > sq ) {
      selected_block = i;
      break;
    }
  }
  HashMap < Integer, Integer > m = new HashMap < > ( );
  for ( int i = 0;
  i <= n;
  i ++ ) {
    if ( ( ( selected_block * sq ) < arr [ i ] ) && ( arr [ i ] <= ( ( selected_block + 1 ) * sq ) ) ) {
      m . put ( arr [ i ], 1 );
      if ( m . get ( arr [ i ] ) == 1 ) return arr [ i ];
    }
  }
  return - 1;
}

File: TransCoder_tokenized_test_set_functions/python/FIND_ONE_MULTIPLE_REPEATING_ELEMENTS_READ_ARRAY.py
def findRepeatingNumber ( arr, n ) :
    sq = sqrt ( n )
    range__ = int ( ( n / sq ) + 1 )
    count = [ 0 for i in range ( range__ ) ]
    for i in range ( 0, n + 1, 1 ) :
        count [ int ( ( arr [ i ] - 1 ) / sq ) ] += 1
    selected_block = range__ - 1
    for i in range ( 0, range__ - 1, 1 ) :
        if ( count [ i ] > sq ) :
            selected_block = i
            break
    m = {
    i : 0 for i in range ( n ) }
    
    for i in range ( 0, n + 1, 1 ) :
        if ( ( ( selected_block * sq ) < arr [ i ] ) and ( arr [ i ] <= ( ( selected_block + 1 ) * sq ) ) ) :
            m [ arr [ i ] ] += 1
            if ( m [ arr [ i ] ] > 1 ) :
                return arr [ i ]
    return - 1
